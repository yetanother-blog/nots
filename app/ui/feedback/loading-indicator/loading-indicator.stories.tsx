import type { Meta, StoryObj } from '@storybook/react';
import { LoadingIndicator } from './loading-indicator';

/** The loading indicator is used to indicate that a process is loading. */
const meta: Meta<typeof LoadingIndicator> = {
  title: '  Feedback/LoadingIndicator',
  component: LoadingIndicator,
};

export default meta;
type Story = StoryObj<typeof LoadingIndicator>;

export const Loading: Story = {
  render: () => <LoadingIndicator />,
};

/**
 * You can also take advantage of the `currentColor` value to inherit the color
 * from the parent element:
 */
export const LoadingWithParent: Story = {
  render: () => (
    <p className="text-nots-primary-500">
      Loading...
      <LoadingIndicator />
    </p>
  ),
};
