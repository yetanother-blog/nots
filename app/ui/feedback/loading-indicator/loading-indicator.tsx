import { clsx } from 'clsx';
import { FC } from 'react';
import { Icon } from '~/ui/data-display/icon/icon';

type LoadingIndicatorProps = {
  className?: string;
} & React.HTMLAttributes<HTMLDivElement>;

export const LoadingIndicator: FC<LoadingIndicatorProps> = ({
  className,
  ...props
}) => {
  return (
    <div
      className={clsx(
        'pointer-events-none flex items-center justify-center',
        className
      )}
      {...props}
    >
      <Icon
        icon="loading"
        className="text-current inline-block h-6 w-6 animate-spin"
        role="status"
        aria-label="loading"
      >
        <span className="sr-only">Loading...</span>
      </Icon>
    </div>
  );
};
