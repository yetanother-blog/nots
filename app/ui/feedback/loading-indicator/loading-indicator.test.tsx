import { render } from '@testing-library/react';
import { LoadingIndicator } from './loading-indicator';

describe('LoadingIndicator Component', () => {
  it('should render', () => {
    const { container } = render(<LoadingIndicator />);
    expect(container).toMatchSnapshot();
  });
});
