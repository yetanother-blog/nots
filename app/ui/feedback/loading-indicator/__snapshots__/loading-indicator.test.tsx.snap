// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingIndicator Component > should render 1`] = `
<div>
  <div
    class="pointer-events-none flex items-center justify-center"
  >
    <svg
      aria-label="loading"
      class="text-current inline-block h-6 w-6 animate-spin"
      fill="transparent"
      focusable="false"
      height="24"
      role="status"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
      <path
        d="M12 2V6M12 18V22M6 12H2M22 12H18M19.0784 19.0784L16.25 16.25M19.0784 4.99994L16.25 7.82837M4.92157 19.0784L7.75 16.25M4.92157 4.99994L7.75 7.82837"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;
